### PR TITLE
Fix naming for multiple resources present in one file

### PR DESCRIPTION
**Pull request by acumino**
**How to categorize this PR?**

/area open-source
/kind enhancement

**What this PR does / why we need it**:
Follow up of - https://github.com/gardener/gardener/pull/8877

If there are multiple resources present in one file the name of the key for data in secret was calculated as `[filename + reosurceName]` but the resource name can contain any character in their name whereas the data key should follow `[-._a-zA-Z0-9]+` regex. 
This PR fixes it by replacing all characters by `_` which doesn't match regex `[-._a-zA-Z0-9]+`.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
Thanks, @plkokanov for reporting it.

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: pkg/chartrenderer/default.go [hunk 1]
```diff
@@ -201,6 +205,8 @@ func (c *RenderedChart) AsSecretData() map[string][]byte {
 				key := strings.ReplaceAll(fileName, "/", "_")
 				if multipleResources {
 					key = strings.ReplaceAll(strings.TrimSuffix(fileName, ".yaml")+"/"+resourceName+".yaml", "/", "_")
+					// replace all characters with _ which does not match the regex [-._a-zA-Z0-9]+
+					key = validConfigKey.ReplaceAllString(key, "_")
```
Comment (plkokanov) [inline pkg/chartrenderer/default.go:209]: Do we need both replaces - the one on line 207 and the one on line 209?

**Review by plkokanov — CHANGES_REQUESTED** [review]
Thanks for the PR. Just one minor question.

File: pkg/chartrenderer/testdata/alpine/templates/alpine-charts.yaml [hunk 2]
```diff
@@ -22,3 +22,17 @@ metadata:
     chartName: {{ .Chart.Name }}
     chartVersion: {{ .Chart.Version | quote }}
 type: Opaque
+---
```
Comment (ialidzhikov) [inline pkg/chartrenderer/testdata/alpine/templates/alpine-charts.yaml]: Please rename the file to something more meaningful - `alpine-resources.yaml` or `alpine.yaml`.

**Review by ialidzhikov — CHANGES_REQUESTED** [review]

**Review by ialidzhikov — COMMENTED** [review]
/lgtm

**Pull request merged**